synapse/rest/media/v1/oembed.py

```diff
@@ -49,8 +49,23 @@ class OEmbedProvider:
     def __init__(self, hs: "HomeServer", client: SimpleHttpClient):
         self._oembed_patterns = {}
         for oembed_endpoint in hs.config.oembed.oembed_patterns:
+            api_endpoint = oembed_endpoint.api_endpoint
+
+            # Only JSON is supported at the moment. This could be declared in
+            # the formats field. Otherwise, if the endpoint ends in .xml assume
+            # it doesn't support JSON.
+            if (
+                oembed_endpoint.formats is not None
+                and "json" not in oembed_endpoint.formats
+            ) or api_endpoint.endswith(".xml"):
+                logger.info(
+                    f"Ignoring oEmbed endpoint due to not supporting JSON: {api_endpoint}"
```

Comment:
```suggestion
                    "Ignoring oEmbed endpoint due to not supporting JSON: %s", api_endpoint
```

Reply:
Why not an f-string here?

Reply:
Because its logging, and using f-strings means that we unconditionally do the string interpolation rather than it being gated behind the log levels. Its not a *huge* deal, but it can add up (especially when we're interpolating variables with more expensive `str` functions)